pandas/core/series.py

```diff
@@ -2539,6 +2539,13 @@ def append(self, to_append, ignore_index=False, verify_integrity=False) -> "Seri
             to_concat.extend(to_append)
         else:
             to_concat = [self, to_append]
+        for x in to_concat[1:]:
+            if isinstance(x, (pd.DataFrame,)):
```

Comment:
```suggestion
            if isinstance(x, ABCDataFrame):
```

Reply:
instead of checking if its a dataframe, would be better to check not an acceptable type, e.g. Series, list-like of Series.

Reply:
IIUC other incorrect types are caught by concat. This PR will be backported so just minimising the changes here to fix the regression (raises AssertionError) for now.

Reply:
could catch the AssertionError and raise improved message instead?

Reply:
no i would rather not backport this, let's actually fix this properly.

Reply:
> no i would rather not backport this, let's actually fix this properly.

and backport the proper fix?

I think we need something backported. to get back to the (dubious) 0.25.3 behaviour could just remove the `self._ensure_type` call added for typing purposes that is causing the AssertionError in 1.0.0rc0.

Reply:
> I think we need something backported. to get back to the (dubious) 0.25.3 behaviour could just remove the self._ensure_type call added for typing purposes that is causing the AssertionError in 1.0.0rc0.

ok let's do that then